src/utils/schema/setLocation.ts

```diff
@@ -1,14 +1,29 @@
-import { Location } from 'src/types';
+import { Location, Place, VirtualLocation } from 'src/types';
 import { setAddress } from './setAddress';
 
 export function setLocation(location: Location) {
-  if (location) {
-    return {
-      ...location,
-      '@type': 'Place',
-      address: setAddress(location.address),
-    };
+  if (!location) {
+    return undefined;
   }
 
-  return undefined;
+  if ('url' in location) {
+    return setVirtualLocation(location);
+  } else {
+    return setPlace(location);
+  }
+}
+
+function setVirtualLocation(location: VirtualLocation) {
```

Comment:
Can we add a Cypress test case to cover for when virtual location is set please.